addons/account/models/chart_template.py

```diff
@@ -578,6 +582,131 @@ def _setup_utility_bank_accounts(self, template_code, company, template_data):
         for company_attr_name, account in zip(accounts_data.keys(), accounts):
             company[company_attr_name] = account
 
+    @api.model
+    def _instantiate_foreign_taxes(self, country, company):
+        """Create and configure foreign taxes from the provided country.
+
+        Instantiate the taxes as they would be for the foreign localization only replacing the accounts used by the most
+        probable account we can retrieve from the company's localization.
+        This method is intended as a shortcut for instantiation, accelerating it, not as an out-of-the-box solution 100%
+        correct solution.
+        """
+        # Implementation:
+        # - Check if there is any tax for this country and stop the process if yes
+        # - Retrieve the tax group and tax template data
+        # - Create accounts at most probable location in the CoA and assign those to fields requiring an account
+        # - Creates tax group and taxes with their ir.model.data
+
+        taxes_in_country = self.env['account.tax'].search([
+            ('country_id', '=', country.id),
+            ('company_id', '=', company.id)
+        ])
+        if taxes_in_country:
+            return
+
+        def create_foreign_tax_account(existing_account, additional_label):
```

Comment:
Yes and no because it's on purpose.
We avoid duplication for the accounts created "based on xml_id" thanks to the caching.
This is done for:
- tax repartition line
- cash basis transition account

For the tax group, we duplicate on purpose. (I can't say why as I just re implemented the feature, I didn't look into why we don't set the same account for all of them)

Reply:
Always challenge everything that isn't obvious :smile: